xbmc/interfaces/legacy/Control.cpp

```diff
@@ -150,7 +150,7 @@ namespace XBMCAddon
     {
       if (!pGUIControl) return NULL;
 
-      LOCKGUI;
+      XBMCAddonUtils::GuiLock(languageHook);
```

Comment:
@Rechi are you okay if I change the PR to your suggestion by explicitly defining the namespace+classname ? It does build indeed fine with that change and I did some testing and did not experience the lockup issue so far.